language/move-ir/types/src/ast.rs

```diff
@@ -392,6 +392,23 @@ pub enum Builtin {
     /// Publish an instantiated struct object into signer's (signer is the first arg) account.
     MoveTo(StructName, Vec<Type>),
 
+    /// Create an empty vector
+    VecEmpty(Vec<Type>),
+    /// Get the length of a vector
+    VecLen(Vec<Type>),
+    /// Acquire an immutable reference to the element at a given index of the vector
+    VecImmBorrow(Vec<Type>),
+    /// Acquire a mutable reference to the element at a given index of the vector
+    VecMutBorrow(Vec<Type>),
+    /// Push an element to the end of the vector
+    VecPushBack(Vec<Type>),
+    /// Pop and return an element from the end of the vector
+    VecPopBack(Vec<Type>),
+    /// Destroy an empty vector
+    VecDestroyEmpty(Vec<Type>),
```

Comment:
IIRC @tnowacki mentioned the idea that instead of having `VecDestroyEmpty`, we add a more generalized `VecDestroyN`. Do we wanna consider this right now, or do we want to do it after the current implementation lands?

Reply:
I like the idea of `VecPack(n)` and `VecUnpack(n)`, but my personal preference would be to implement it in a separate PR.

Reply:
Could you just remove VecDestroyEmpty and VecEmpty from this PR then? feels weird to land them just to change them

Reply:
Hmm, that can be a bit ugly I guess. Removing `VecDestroyEmpty` and `VecEmpty` will break some of the IR tests because these tests depends on `VecEmpty` to produce a valid vector.

Reply:
Then you can continue to use the native functions no? Those aren't going anywhere anytime soon

Reply:
Yes, but need to additionally import the `0x1.Vector;`. Not a big deal though, but just some extra work. If this is the route, I think maybe adding `VecPack(n)` and `VecUnpack(n)` in this PR would make more sense.

Reply:
Either way is fine by me :) I think the size of the review will be roughly the same either way so whatever is easier for you

Reply:
I would prefer to have this landed first :) It has been here for so long and rebased for so many times :)

Reply:
Sure... I get that, just very awkward to add bytecode operations just to remove them.

Reply:
Yup, I understand the awkwardness. I owe a big thank you to you and whomever else that helped reviewing this giant PR.

Reply:
@tnowacki @vgao1996 This PR is now ported to using the `VecPack(type, num)` and `VecUnpack(type, num)` bytecode (the porting is done in the last three commits).

`VecEmpty` and `VecDestroyEmpty` are gone as they are now just special cases of `VecPack(type, 0)`, `VecUnpack(type, 0)` respectively.